torch/_prims/__init__.py

```diff
@@ -1912,3 +1838,143 @@ def _make_bool_reduction_prim(name: str, impl_aten, doc):
     impl_aten=torch.any,
     doc="",
 )
+
+# TODO: layout, pin_memory, memory_format
+# TODO: model requires_grad on TensorMeta
+def _empty_meta(
+    shape: ShapeType, *, dtype: torch.dtype, device: torch.device, requires_grad: bool
+) -> TensorLikeType:
+    strides = utils.make_contiguous_strides_for(shape)
+    return TensorMeta(shape=shape, strides=strides, dtype=dtype, device=device)
+
+
+def _empty_aten(
+    shape: ShapeType, *, dtype: torch.dtype, device: torch.device, requires_grad: bool
+) -> Tensor:
+    return torch.empty(shape, dtype=dtype, device=device, requires_grad=requires_grad)
+
+
+_empty_doc = """
+    Creates a tensor with uninitialized values and the specified shape, dtype, and device.
+"""
+
+empty = _make_prim(
+    schema="empty(int[] shape, *, ScalarType dtype, Device device, bool requires_grad) -> Tensor",
```

Comment:
Should empty really have a `requires_grad` argument in PrimTorch? From the perspective of a backend implementer `requires_grad` ought to have been long erased; there's nothing they're going to be usefully able to do with it.

Reply:
It's certainly interesting to consider;  we can always make it exclusive to the ref later when we get into autograd